### DISCUSSION TITLE
upgrading from a very old version of symfony

### DISCUSSION THREAD

**Discussion by dabeegmon** [symfony/symfony#45120] [Q&A]
greetings

a project that's not exactly current - - - is there a somewhat easy way to upgrade from symfony 2.0 to 6.xx?

**Comment by GromNaN** [comment]
Depending on how much the project is tighed to the components that changed, I see 2 approches:

1. Upgrade to major versions, one-by-one (from 2.x to 2.8, then 2.8 to 3.4, then 3.4 to 4.4, then 4.4 to 5.4, then 6.0). There is an UPGRADE file in each version. You will have to modify the same files several time. And the directory structure have changed several times.
2. Or create a new fresh project and rebuild your project by importing your files (each module or feature). symfony/flex will help to boostrap the configuration of the principal bundles.

You can use Rector and Symplify (See [symplify/config-transformer](https://github.com/symplify/config-transformer)) to help.

I hope you have tests.

**Comment by dabeegmon** [comment]
On Sat, Jan 22, 2022 at 3:35 PM Jérôme Tamarelle ***@***.***>
wrote:

> Depending on how much the project is tighed to the components that
> changes, I see 2 approches:
>
>    1. Upgrade to major versions, one-by-one (from 2.x to 2.8, then 2.8 to
>    3.4, then 3.4 to 4.4, then 4.4 to 5.4, then 6.0). There is an UPGRADE file
>    in each version. You will have to modify the same files several time. And
>    the directory structure have changed several times.
>    2. Or create a new fresh project and rebuild your project by importing
>    your files (each module or feature). symfony/flex will help to boostrap the
>    configuration of the principal bundles.
>
> I hope you have tests.
>
>
> Hmmmmm - - - interesting.

Thanking you for your assistance.

Regards

**Comment by helmis123** [comment]
I think it's better to create a fresh new project and rebuild your project by importing your files.
